sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/jdbc/JdbcConnectionString.java

```diff
@@ -68,69 +81,128 @@ private void resolveSegments() {
                 throw new IllegalArgumentException(String.format(INVALID_PROPERTY_PAIR_FORMAT, tokenValuePair));
             }
             if (pair.length < 2) {
-                this.properties.put(key, NONE_VALUE);
+                properties.put(key, null);
             } else {
-                this.properties.put(key, pair[1]);
+                properties.put(key, pair[1]);
             }
+            originalPropertiesOrder.add(key);
         }
+        return new JdbcConnectionString(originalJdbcURL, baseURL, databaseType, properties, originalPropertiesOrder);
     }
 
-    public String enhanceConnectionString(Map<String, String> enhancedProperties) {
-        if (enhancedProperties == null || enhancedProperties.isEmpty()) {
-            return this.jdbcURL;
+    public String getJdbcUrl() {
+        if (this.enhancedProperties.isEmpty()) {
+            return this.originalJdbcUrl;
         }
-        LOGGER.debug("Trying to enhance jdbc url for {}", databaseType);
 
-        StringBuilder builder = new StringBuilder(this.jdbcURL);
+        LOGGER.debug("Trying to construct enhanced jdbc url for {}", databaseType);
 
-        if (!this.hasProperties()) {
-            builder.append(databaseType.getPathQueryDelimiter());
-        } else {
-            builder.append(databaseType.getQueryDelimiter());
-        }
+        StringBuilder builder = new StringBuilder(this.baseUrl).append(databaseType.getPathQueryDelimiter());
+
+        Map<String, String> mergedProperties = new TreeMap<>(originalProperties);
+        mergedProperties.putAll(this.enhancedProperties);
+
+        this.orderedOriginalPropertyKeys.forEach(k -> builder
+            .append(constructPropertyString(k, mergedProperties.remove(k)))
+            .append(databaseType.getQueryDelimiter()));
+
+        mergedProperties.forEach((k, v) -> builder.append(k).append("=").append(v).append(databaseType.getQueryDelimiter()));
+
+        String enhancedUrl = builder.toString();
+        return enhancedUrl.substring(0, enhancedUrl.length() - 1);
```

Comment:
I don't understand the reason, could you explain it here?

Reply:
remove the last query delimiter.